src/test/java/no/entur/antu/validation/validator/servicejourney/passingtime/NonIncreasingTimetabledPassingTimeValidatorTest.java

```diff
@@ -1,502 +1,589 @@
-          NonIncreasingPassingTimeValidator.RULE_INCOMPLETE_TIME.name()
-        ),
-      is(true)
+    NonIncreasingPassingTimeValidator validator =
+      new NonIncreasingPassingTimeValidator();
+    List<ValidationIssue> issues = validator.validate(
+      netexTestEnvironment.getJaxbValidationContext()
+    );
+    Assertions.assertTrue(
+      containsValidationIssuesForRule(issues, RULE_INCONSISTENT_TIME)
     );
   }
 
   @Test
-  void testValidateServiceJourneyWithAreaStopInconsistentTimeWindow() {
-    Context context = Context.create();
-    // remove arrival time and departure time and add flex window
-    context.timetabledPassingTimes
-      .get(0)
-      .withDepartureTime(null)
-      .withArrivalTime(null)
-      .withEarliestDepartureTime(LocalTime.MIDNIGHT.plusMinutes(1))
-      .withLatestArrivalTime(LocalTime.MIDNIGHT);
-
-    NetexEntitiesIndex netexEntitiesIndex =
-      context.netexEntitiesTestFactory.create();
+  void testValidateServiceJourneyWithFlexibleStop() {
+    ServiceJourney serviceJourney =
+      NetexTestDataSample.serviceJourneyWithFlexibleTimetabledPassingTime();
+    JourneyPattern journeyPattern = NetexTestDataSample.defaultJourneyPattern();
+    FlexibleStopPlace flexibleStopPlace =
+      NetexTestDataSample.defaultFlexibleStopPlace();
+
+    NetexTestEnvironment baseTestEnvironment = buildNetexTestEnvironment(
+      serviceJourney,
+      journeyPattern
+    );
 
-    netexEntitiesIndex
-      .getFlexibleStopPlaceIdByStopPointRefIndex()
-      .put(context.scheduledStopPointRefs.get(0).getRef(), "");
+    NonIncreasingPassingTimeValidator validator =
+      new NonIncreasingPassingTimeValidator();
 
-    mockGetStopName(
-      ScheduledStopPointId.of(context.scheduledStopPointRefs.get(0))
+    NetexTestEnvironment envWithFlexibleStopFromSharedFile =
+      addFlexStopPlaceFromSharedFileToNetexTestEnvironment(
+        baseTestEnvironment,
+        flexibleStopPlace
+      );
+    List<ValidationIssue> issuesWhenUsingSharedFile = validator.validate(
+      envWithFlexibleStopFromSharedFile.getJaxbValidationContext()
+    );
+    Assertions.assertFalse(
+      containsValidationIssuesForRule(
+        issuesWhenUsingSharedFile,
+        RULE_INCOMPLETE_TIME
+      )
     );
-    ValidationReport validationReport = runValidation(netexEntitiesIndex);
 
-    assertThat(validationReport.getValidationReportEntries().size(), is(1));
-    assertThat(
-      validationReport
-        .getNumberOfValidationEntriesPerRule()
-        .containsKey(
-          NonIncreasingPassingTimeValidator.RULE_INCONSISTENT_TIME.name()
-        ),
-      is(true)
+    NetexTestEnvironment envWithFlexibleStopFromLineFile =
+      addFlexStopPlaceFromLineFileToNetexTestEnvironment(
+        baseTestEnvironment,
+        flexibleStopPlace
+      );
+    List<ValidationIssue> issuesWhenUsingLineFile = validator.validate(
+      envWithFlexibleStopFromLineFile.getJaxbValidationContext()
+    );
+    Assertions.assertFalse(
```

Comment:
could you make the code more concise by introducing a helper method` assertNoIssueForRule(...)` ?